### DISCUSSION TITLE
want to contribute the Hausdorff distance

### DISCUSSION THREAD

**Discussion by jakubMitura14** [Project-MONAI/MONAI#3919] [Ideas]
Hello I developed pretty decent CUDA 11.5  C++ algorithm for Hausdorff distance computation in 3D setting , that seem to be faster then current implementation. I would like to contribute it.

Hence I am trying to compile the pyTorch CUDA extension to conduct more tests , Hovewer I repeatadly fail in this for days. 
I had seen That MONAI has dockerfile - is it good idea to try to develop CUDA extension from Docker (I have nearly 0 experience with docker - but as far as I understand It should make the configuration simplified).

**Comment by wyli** [comment]
cool,  the docker image is only one of MONAI's release channels, ideally the c++ code should be built so that the users can download/compile in this way: https://docs.monai.io/en/latest/installation.html#from-github 

perhaps you can follow this lltm example https://github.com/Project-MONAI/MONAI/blob/dev/monai/csrc/ext.cpp#L26-L28 or go for the jit approach https://github.com/Project-MONAI/MONAI/blob/dev/monai/_extensions/loader.py please feel free to submit a draft PR, we can help integrate it into the core codebase. cc @charliebudd

**Reply by jakubMitura14** [reply]
Fantastic! so I will work on the examples that you had shared, thanks !
